v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -0,0 +1,647 @@
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, newTaskDefinition.id);
+        assert.deepEqual(usedCliArguments, [true, true, true]);
+        assert.deepEqual(res.taskArguments, { optionFlag: "<value>" });
+      });
+
+      it("should throw because the parameter is not defined", async function () {
+        const command = "npx hardhat task --undefinedParam <value>";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        assert.throws(
+          () => parseTaskAndArguments(cliArguments, usedCliArguments, hre),
+          new HardhatError(
+            HardhatError.ERRORS.ARGUMENTS.UNRECOGNIZED_NAMED_PARAM,
+            {
+              parameter: "--undefinedParam",
+            },
+          ),
+        );
+      });
+
+      it("should throw because the task parameter is declared before the task name", async function () {
+        const command = "npx hardhat --param <paramValue> task";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        assert.throws(
+          () => parseTaskAndArguments(cliArguments, usedCliArguments, hre),
+          new HardhatError(
+            HardhatError.ERRORS.ARGUMENTS.UNRECOGNIZED_NAMED_PARAM,
+            {
+              parameter: "--param",
+            },
+          ),
+        );
+      });
+
+      it("should throw because the task parameter is declared but no value is associated to it", async function () {
+        const command = "npx hardhat task --param";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false];
+
+        assert.throws(
+          () => parseTaskAndArguments(cliArguments, usedCliArguments, hre),
+          new HardhatError(
+            HardhatError.ERRORS.ARGUMENTS.MISSING_VALUE_FOR_NAMED_PARAMETER,
+            {
+              paramName: "--param",
+            },
+          ),
+        );
+      });
+    });
+
+    describe("task and subtask with positional parameters", function () {
+      before(async function () {
+        newTaskDefinition = task(["task"])
+          .addPositionalParameter({ name: "param" })
+          .addPositionalParameter({
+            name: "param2",
+            defaultValue: "defaultValue",
+          })
+          .setAction(() => {})
+          .build();
+
+        newSubtaskDefinition = task(["task", "subtask"])
+          .addPositionalParameter({ name: "param" })
+          .addPositionalParameter({
+            name: "param2",
+            defaultValue: "defaultValue",
+          })
+          .setAction(() => {})
+          .build();
+
+        hre = await createHardhatRuntimeEnvironment({
+          tasks: [newTaskDefinition, newSubtaskDefinition],
+        });
+      });
+
+      it("should get the tasks and its required parameter", function () {
```

Comment:
what if it's missing?

Reply:
This comment made me realize that I forgot to add a function to check the required parameters, I implemented it and added the tests in this commit: https://github.com/NomicFoundation/hardhat/pull/5300/commits/2b9eb55b0404b9a700c86981063a5223c6de88d4